File: sdks/csharp/src/NovaBilling/PaymentProviders/Requests/UpdateProviderDto.cs
using System.Text.Json.Serialization;
using NovaBilling.Core;

namespace NovaBilling;

[Serializable]
public record UpdateProviderDto
{
    /// <summary>
    /// Payment provider ID
    /// </summary>
    [JsonIgnore]
    public required string Id { get; set; }

    /// <summary>
    /// Provider name
    /// </summary>
    [JsonPropertyName("providerName")]
    public string? ProviderName { get; set; }

    /// <summary>
    /// Provider credentials (will be encrypted)
    /// </summary>
    [JsonPropertyName("credentials")]
    public Dictionary<string, object?>? Credentials { get; set; }

    [JsonPropertyName("isActive")]
    public bool? IsActive { get; set; }

    /// <summary>
    /// Priority (lower = higher)
    /// </summary>
    [JsonPropertyName("priority")]
    public double? Priority { get; set; }

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}

File: sdks/rust/src/api/types/update_provider_dto.rs
pub use crate::prelude::*;

/// Request type for API operation
#[derive(Debug, Clone, Serialize, Deserialize, Default, PartialEq)]
pub struct UpdateProviderDto {
    /// Provider name
    #[serde(rename = "providerName")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub provider_name: Option<String>,
    /// Provider credentials (will be encrypted)
    #[serde(skip_serializing_if = "Option::is_none")]
    pub credentials: Option<HashMap<String, serde_json::Value>>,
    #[serde(rename = "isActive")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub is_active: Option<bool>,
    /// Priority (lower = higher)
    #[serde(skip_serializing_if = "Option::is_none")]
    pub priority: Option<f64>,
}
